### PR TITLE
improve aggregations perf

### PR DESCRIPTION
**Pull request by syl10100**
**Description:**

1. create products_tags collection with script refresh_products_tags.js
2. modify files Data.pm and Display.pm to use products_tags collection in aggregate.
3. schedule refresh of products_tags collection

**Related issues and discussion:** #1610

### CODE REVIEW

File: scripts/refresh_products_tags.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/* 
+  Name        : refresh_products_tags.js 
+  Description : Refresh products_tags collection by copying *_tags fields from products collection 
+  Version     : 1.0
+  Usage       : mongo refresh_products_tags.js
+  Add in crontab for daily refresh : 
+	00 02 * * * /usr/bin/mongo /etc/mongo/refresh_products_tags.js  >> /var/log/mongodb/refresh_products_tags.log
+*/
+
+db = db.getSiblingDB('off');
```
Comment (stephanegigandet) [inline scripts/refresh_products_tags.js]: is there a way to make the db name a parameter? we have separate names for OBF, OPF, OPFF

File: scripts/refresh_products_tags.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	codes_tags:1
+	}},
+{"$out": "products_tags"}
+]);
+
+print(Date() + ' : ' + db.products_tags.estimatedDocumentCount() + ' products refreshed in products_tags collection.');
+
+
+print(Date() + ' : Create indexes for products_tags collection (if not already existing)...');
+db.products_tags.createIndex({countries_tags: 1});
```
Comment (stephanegigandet) [inline scripts/refresh_products_tags.js]: isn't that going to block the database without the background : true ?

**Comment by stephanegigandet** [comment]
It looks good to me, thank you. I'll probably change it a little to make it optional, so that we can test it more easily.

**Comment by syl10100** [comment]
> 
> 
> It looks good to me, thank you. I'll probably change it a little to make it optional, so that we can test it more easily.

Yes good idea

**Pull request merged**